### PR TITLE
Feature: BB-210 handle readiness liveness oplogPopulator

### PR DESCRIPTION
**Pull request by welansari**

### CODE REVIEW

File: extensions/oplogPopulator/OplogPopulatorTask.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 const OplogPopulator = require('./OplogPopulator');
 
+const {
+    DEFAULT_LIVE_ROUTE,
+    DEFAULT_READY_ROUTE,
+} = require('arsenal').network.probe.ProbeServer;
+const { sendSuccess, sendError } = require('arsenal').network.probe.Utils;
+const { startProbeServer } = require('../../lib/util/probe');
+
+const startProbeServerAsync = util.promisify(startProbeServer);
```
Comment (miniscruff) [inline extensions/oplogPopulator/OplogPopulatorTask.js]: Would you mind adding this util to the /lib/util/probe file so other probe servers can use it?
Note: Not sure if we should use the async suffix or a promise suffix, but picking one and using it across the board is ideal

File: extensions/oplogPopulator/OplogPopulatorConfigValidator.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 const joi = require('joi');
+const { probeServerJoi } = require('../../lib/config/configItems.joi.js');
```
Comment (jbertran) [inline extensions/oplogPopulator/OplogPopulatorConfigValidator.js]: We can let node work out the extension.

```suggestion
const { probeServerJoi } = require('../../lib/config/configItems.joi');
```

**Review by jbertran — APPROVED** [review]

File: lib/util/probe.js [hunk 3]
```diff
@@ -33,6 +34,14 @@ function startProbeServer(config, callback) {
     probeServer.start();
 }
 
+/**
+ * Start probe server for Queue Processor
+ * @param {ProbeServerConfig} config - Configuration for probe server
+ * @returns {Promise|ProbeServer} Probe server or undefined if disabled
+ */
+const startProbeServerPromise = util.promisify(startProbeServer);
```
Comment (francoisferrand) [inline lib/util/probe.js:42]: not sure introducing this method makes the code more readable...

A better way may be to update the `startProbeServer` function to return a promise when no callback is provided:

```
// @return {Promise|undefined}
function startProbeServer(config, callback) {
   if (callback == undefined)
     return util.promisify(startProbeServer)(config);

  // else return legacy code
  [...]
}

**Review by francoisferrand — APPROVED** [review]

**Pull request merged**